editor.planx.uk/src/@planx/components/MultipleFileUpload/Public.test.tsx

```diff
@@ -208,3 +222,133 @@ describe("Modal trigger", () => {
     expect(fileTaggingModal).not.toBeVisible();
   });
 });
+
+describe("Adding tags and syncing state", () => {
+  test("Can continue when all required file types are uploaded and tagged", async () => {
+    const handleSubmit = jest.fn();
+    const { user } = setup(
+      <MultipleFileUploadComponent
+        title="Test title"
+        handleSubmit={handleSubmit}
+        fileTypes={mockFileTypesUniqueKeys}
+      />
+    );
+
+    // No file requirements have been satisfied yet
+    let incompleteIcons = screen.getAllByTestId("incomplete-icon");
+    expect(incompleteIcons).toHaveLength(3);
+
+    // Upload one file
+    mockedAxios.post.mockResolvedValueOnce({
+      data: {
+        file_type: "image/png",
+        fileUrl: "https://api.editor.planx.dev/file/private/gws7l5d1/test1.jpg",
+      },
+    });
+
+    const file1 = new File(["test1"], "test1.png", { type: "image/png" });
+    const input = screen.getByTestId("upload-input");
+    await user.upload(input, [file1]);
+
+    // Modal opened automatically
+    const fileTaggingModal = await within(document.body).findByTestId(
+      "file-tagging-dialog"
+    );
+
+    // The number of selects in the modal matches the number of uploaded files
+    const selects = await within(document.body).findAllByTestId("select");
+    expect(selects).toHaveLength(1);
+
+    // Apply multiple tags to this file
+    fireEvent.change(selects[0], { target: { value: "Roof plan" } });
```

Comment:
I'm guessing this is a necessary compromise as without `native={true}` we can't use something like `user.selectOptions()`?

Reply:
exactly - even with `native={true}` which I fiddled with locally for awhile `user.selectOptions()` still won't work exactly as expected unless we refactor our <options> to be a more true list of just checkboxes without list subheaders which was a pretty crucial part of the design / we've already comprised a bit there going from 2-lines to one.